File: src/main/java/com/dambra/paul/stringcalculator/calculationParts/BareNumber.java
package com.dambra.paul.stringcalculator.calculationParts;

public class BareNumber extends CalculationPart {
    private final double value;

    public BareNumber(String value) {
        this.value = Double.valueOf(value);
    }

    @Override
    public double Calculate(double total) {
        return next.Calculate(total + value);
    }
}

File: src/main/java/com/dambra/paul/stringcalculator/calculationParts/BareNumber.kt
package com.dambra.paul.stringcalculator.calculationParts

class BareNumber(value: String) : CalculationPart() {
    private val value: Double = java.lang.Double.valueOf(value)

    override fun calculate(total: Double) =
            next.calculate(total + value)
}
